src/redisearch_rs/CMakeLists.txt

```diff
@@ -0,0 +1,105 @@
+cmake_minimum_required(VERSION 3.13)
+
+# Find Rust and Cargo
+find_program(CARGO_EXECUTABLE cargo REQUIRED)
+
+if(NOT CARGO_EXECUTABLE)
+    message(FATAL_ERROR "Cargo not found. Please install Rust and Cargo.")
+endif()
+
+# Determine Rust profile - allow invoker to set it, but have fallback logic if not set
+if(NOT DEFINED RUST_PROFILE)
+    set(RUST_PROFILE "release")
+    if(DEFINED BUILD_SEARCH_UNIT_TESTS AND BUILD_SEARCH_UNIT_TESTS)
+        if(CMAKE_BUILD_TYPE STREQUAL "Debug" OR DEFINED SAN OR DEFINED COV)
+            set(RUST_PROFILE "dev")
+        else()
+            set(RUST_PROFILE "optimised_test")
+        endif()
+    else()
+        # For non-test builds, only use dev profile for Debug builds (not SAN/COV)
+        if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+            set(RUST_PROFILE "dev")
+        endif()
+    endif()
+endif()
+
+# Set up RUSTFLAGS - respect existing environment variable and default to -D warnings
+if(DEFINED ENV{RUSTFLAGS})
+    set(RUST_FLAGS "$ENV{RUSTFLAGS}")
+else()
+    set(RUST_FLAGS "-D warnings")
+endif()
```

Comment:
We should default to empty. `-D warnings` only makes sense for project developers.